# alc_multiomics

Integration of “multi-omics” data is often necessary to nominate candidate causal variants and genes and prioritize them for follow up studies. Here, we used Mendelian randomization to integrate AUD and drinks per week (DPW) GWAS summary statistics with the gene expression and methylation quantitative trait loci (eQTLs and mQTLs) in the largest brain and myeloid datasets. We also used AUD-related single cell epigenetic data to nominate candidate causal variants and genes associated with DPW and AUD.

Results of the analysis can be explored at our Shniy web app (https://lcad.shinyapps.io/alc_multiomics/).

Scripts and pipelines used in the analyes are deposited at this GitHub repository. We will also make full summary statistics downloadable after our primary manuscript will be accepted. Currently the associations that passed significance threshold in our SMR analysis can be accessed at this page.
